src/peft/utils/merge_utils.py

```diff
@@ -0,0 +1,237 @@
+# coding=utf-8
+# Copyright 2024-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from typing import Literal
+
+import torch
+
+
+def reshape_weight_task_tensors(task_tensors, weights):
+    new_shape = weights.shape + (1,) * (len(task_tensors.shape) - len(weights.shape))
```

Comment:
`len(task_tensors.shape)` could be simplified to `task_tensors.dim()`, right? Same for `weights`. Could you please also add a short docstring or comment to indicate what this function does?